openraft/src/engine/engine.rs

```diff
@@ -145,10 +146,11 @@ impl<NID: NodeId> Engine<NID> {
         }
         self.try_update_membership(entry);
 
-        // TODO: set target state. This should be done by Engine but currently it is not.
-
         self.commands.push(Command::MoveInputCursorBy { n: l });
 
+        // With the new config, start to elect to become leader
```

Comment:
I can get the point: why the node init state is candidate instead of follower?

Reply:
it means if the node startup in whatever situation, even if it join a cluster, it must start an election?

Reply:
After initializing the cluster(by appending a membership log), the cluster should elect a leader ASAP.
The behavior for a **follower** is to wait for a timeout before starting elect(by entering **candidate** state). Thus we can just avoid the **follower** state and enter **candidate** state at once.

Reply:
> it means no matter the node startup in whatever situation, even if it join a cluster, it must start an election?

It must be a mistake to join a node into an existing cluster by `initialize()` it.
To join, it does not call `initialize()`.

It should send a `change-membership` RPC to the existing cluster leader to start replicating logs to the new node.

Reply:
> it means no matter the node startup in whatever situation, even if it join a cluster, it must start an election?

BTW, the election will never win. The new node does not have any log on it thus it will be rejected.

Reply:
seems to me that init as a follower or a candidate has not too much different, they all have advantages and disadvantages, as you like:)